api/v1alpha1/extension_types.go

```diff
@@ -0,0 +1,169 @@
+	// name specifies the name of the name of the package
+	Name string `json:"name"`
+
+	//+kubebuilder:validation:MaxLength:=64
+	//+kubebuilder:validation:Pattern=`^(\s*(=||!=|>|<|>=|=>|<=|=<|~|~>|\^)\s*(v?(0|[1-9]\d*|[x|X|\*])(\.(0|[1-9]\d*|x|X|\*]))?(\.(0|[1-9]\d*|x|X|\*))?(-([0-9A-Za-z\-]+(\.[0-9A-Za-z\-]+)*))?(\+([0-9A-Za-z\-]+(\.[0-9A-Za-z\-]+)*))?)\s*)((?:\s+|,\s*|\s*\|\|\s*)(=||!=|>|<|>=|=>|<=|=<|~|~>|\^)\s*(v?(0|[1-9]\d*|x|X|\*])(\.(0|[1-9]\d*|x|X|\*))?(\.(0|[1-9]\d*|x|X|\*]))?(-([0-9A-Za-z\-]+(\.[0-9A-Za-z\-]+)*))?(\+([0-9A-Za-z\-]+(\.[0-9A-Za-z\-]+)*))?)\s*)*$`
+	//+kubebuilder:Optional
+	// Version is an optional semver constraint on the package version. If not specified, the latest version available of the package will be installed.
+	// If specified, the specific version of the package will be installed so long as it is available in any of the content sources available.
+	// Examples: 1.2.3, 1.0.0-alpha, 1.0.0-rc.1
+	//
+	// For more information on semver, please see https://semver.org/
+	// version constraint definition
+	Version string `json:"version,omitempty"`
+
+	//+kubebuilder:validation:MaxLength:=48
+	//+kubebuilder:validation:Pattern:=^[a-z0-9]+([\.-][a-z0-9]+)*$
+	// channel constraint definition
+	Channel string `json:"channel,omitempty"`
+
+	//+kubebuilder:validation:Enum:=Enforce;Ignore
+	//+kubebuilder:default:=Enforce
+	//+kubebuilder:Optional
+	//
+	// upgradeConstraintPolicy Defines the policy for how to handle upgrade constraints
+	UpgradeConstraintPolicy UpgradeConstraintPolicy `json:"upgradeConstraintPolicy,omitempty"`
+}
+
+// TODO: Implement ExtensionSourceDirect containing a URL or other reference mechanism
+
+type ExtensionSource struct {
+	// A source package defined by a name, version and/or channel
+	Package *ExtensionSourcePackage `json:"package,omitempty"`
+
+	// A direct package defined by a URL
+	// TODO: Direct *ExtensionSourceDirect `json:"direct,omitempty"`
+}
+
+// ExtensionSpec defines the desired state of Extension
+type ExtensionSpec struct {
+	//+kubebuilder:validation:Enum:=Active;Paused
+	//+kubebuilder:default:=Active
+	//+kubebuilder:Optional
+	//
+	// managed controls the management state of the extension. "Active" means this extension will be reconciled and "Paused" means this extension will be ignored.
+	Managed ExtensionManagedState `json:"managed,omitempty"`
+
+	//+kubebuilder:validation:MaxLength:=253
+	//+kubebuilder:validation:Pattern:=^[a-z0-9]+([\.-][a-z0-9]+)*$
+	//
+	// serviceAccountName is he name of a service account in the Extension's namespace that will be used to manage the installation and lifecycle of the extension.
+	ServiceAccountName string `json:"serviceAccountName"`
+
+	//+kubebuilder:validation:MaxLength:=253
+	//+kubebuilder:validation:Pattern:=^[a-z0-9]+([\.-][a-z0-9]+)*$
+	//+kubebuilder:Optional
+	//
+	// defaultNamespace is the location of installation if different than the resource namespace
+	DefaultNamespace string `json:"defaultNamespace,omitempty"`
```

Comment:
I still vote that we drop this for now. We already have existing logic for namespace defaulting in OLMv0, which I think we would hardcode into the YTT template we use?

If/when we care to support other bundle types, we can revisit.

Reply:
This is currently unused in the code I have, so not a problem.